python/ray/train/callbacks/results_preprocessors/aggregate.py

```diff
@@ -0,0 +1,108 @@
+import logging
+from typing import Dict, List, Optional
+
+import numpy as np
+
+from ray.train.callbacks.results_preprocessors.preprocessor import ResultsPreprocessor
+from ray.train.callbacks.results_preprocessors.aggregate_fn import (
+    AggregateFn,
+    Average,
+    Max,
+    WeightedAverage,
+    VALID_AGGREGATE_TYPES,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def _get_values(key, reported_metrics, results):
+    """Return values in the results list from all workers."""
+    if key not in reported_metrics:
+        logger.warning(
+            f"`{key}` is not reported from workers, so it is ignored. "
+            "Please make sure that it is saved using `train.report()`."
+        )
+        return []
```

Comment:
Suggestion: Use `None` to represent the failure case instead of empty list so that it is more clear (as an example, to differentiate from when `results` is empty).